File: 2025-1/Basic/SJP/Stack/BOJ_10828.cpp
#include <iostream>
#include <stack>
using namespace std;

int main(){
    int N;
    cin >> N;
    stack<int> stack;
    while (N > 0){
        string text;
        cin >> text;
        if (text == "push"){
            int X;
            cin >> X;
            stack.push(X);
        }
        else if (text == "pop"){
            if (stack.size() == 0){
                cout << -1 << endl;
            }
            else{
                cout << stack.top() << endl;
                stack.pop();
            }
        }
        else if (text == "size"){
            cout << stack.size() << endl;
        }
        else if (text == "empty"){
            if (stack.size() == 0) cout << 1 << endl;
            else cout << 0 << endl;
        }
        else if (text == "top"){
            if (stack.size() == 0) cout << -1 << endl;
            else cout << stack.top() << endl;
        }
        N--;
    }
}

File: 2025-1/Basic/SJP/Stack/BOJ_10828.py
import sys
input = sys.stdin.readline
stack = []
for _ in range(int(input())):
    O = [*input().split()]
    if O[0] == 'push':
        stack.append(O[1])
    elif O[0] == 'pop':
        if len(stack)!=0:
            sys.stdout.write(stack.pop() + '\n')
        else:
            sys.stdout.write('-1' + '\n')
    elif O[0] == 'size':
        sys.stdout.write(str(len(stack)) + '\n')
    elif O[0] == 'empty':
        if len(stack) == 0:
            sys.stdout.write('1' + '\n')
        else:
            sys.stdout.write('0' + '\n')
    elif O[0] == 'top':
        if len(stack) != 0:
            sys.stdout.write(stack[-1] + '\n')
        else:
            sys.stdout.write('-1' + '\n')
